ksqldb-engine/src/main/java/io/confluent/ksql/util/SharedKafkaStreamsRuntimeImpl.java

```diff
@@ -221,6 +205,35 @@ public void stop(final QueryId queryId, final boolean resetOffsets) {
     log.info("Query {} was stopped successfully", queryId);
   }
 
+  private void tryCleanupTopology(QueryId queryId, boolean resetOffsets) {
```

Comment:
Hey this is much better. I am concerned with line 201. If it fails after 3 tries we should not execute this code.

```    
if (resetOffsets) {
      // we don't want to lose it from this runtime
      collocatedQueries.remove(queryId);
}

Reply:
Thanks for taking a look William!
After 3 retries we would have already thrown an Exception at line 225 right?